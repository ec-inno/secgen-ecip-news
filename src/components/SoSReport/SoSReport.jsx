import React from 'react';
import PropTypes from 'prop-types';
import { useTranslation } from 'react-i18next';

import getCountries from '@eci/utils/getCountries';

const SoSReport = ({ submission }) => {
  if (!submission) return '';

  const hasEntries = submission.entry && submission.entry.length !== 0;
  const hasTotalSignatures = Boolean(
    submission.totalSignatures && Number(submission.totalSignatures) > 1
  );

  if (!hasEntries && !hasTotalSignatures) return '';

  const { t } = useTranslation();
  const countries = getCountries(t);

  return (
    <>
      <h2 className="ecl-u-type-heading-2">
        {t(
          'Number of statements of support collected as reported by the organisers'
        )}
      </h2>
      <table className="eci-table ecl-u-type-paragraph">
        <tbody>
          {hasEntries &&
            submission.entry.map((e, key) => {
              const code = e.countryCodeType.toLowerCase();
              const country = countries[code] || code;

              return (
                <tr key={key}>
                  <td>{country}</td>
                  <td>
                    {e.afterSubmission ? `${e.total}<sup>*</sup>` : e.total}
                  </td>
                </tr>
              );
            })}
          {hasTotalSignatures && (
            <tr className="ecl-u-type-bold">
              <td>{t('Total number of signatories')}</td>
              <td>{submission.totalSignatures}</td>
            </tr>
          )}
        </tbody>
      </table>
      <p className="ecl-u-type-paragraph-s">
        *{' '}
        {t(
          'Certificates received after the time of submission. Figures not included in the total number of signatories.'
        )}
      </p>
    </>
  );
};

SoSReport.propTypes = {
  submission: PropTypes.shape({
    totalSignatures: PropTypes.number,
    entry: PropTypes.arrayOf(
      PropTypes.shape({
        countryCodeType: PropTypes.string,
        total: PropTypes.number,
        afterSubmission: PropTypes.bool,
      })
    ),
  }),
};

export default SoSReport;
