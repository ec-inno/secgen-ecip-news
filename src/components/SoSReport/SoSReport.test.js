import React from 'react';
import renderer from 'react-test-renderer';

import SoSReport from './SoSReport';

describe('SoSReport', () => {
  it('no input: returns an empty string', () => {
    const tree = renderer.create(<SoSReport />).toJSON();
    expect(tree).toBe('');
  });

  it('requires more than 1 signature by totalSignatures: case 0', () => {
    const props = {
      submission: {
        totalSignatures: 0,
      },
    };

    const tree = renderer.create(<SoSReport {...props} />).toJSON();
    expect(tree).toBe('');
  });

  it('requires more than 1 signature by totalSignatures: case 1', () => {
    const props = {
      submission: {
        totalSignatures: 1,
      },
    };

    const tree = renderer.create(<SoSReport {...props} />).toJSON();
    expect(tree).toBe('');
  });

  it('renders signatures information when greater than 1 by totalSignatures', () => {
    const props = {
      submission: {
        totalSignatures: 2,
      },
    };

    const tree = renderer.create(<SoSReport {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders entries when provided', () => {
    const props = {
      submission: {
        entry: [
          {
            countryCodeType: 'BG',
            total: 12598,
            afterSubmission: false,
          },
          {
            countryCodeType: 'DK',
            total: 4610,
            afterSubmission: false,
          },
        ],
      },
    };

    const tree = renderer.create(<SoSReport {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
