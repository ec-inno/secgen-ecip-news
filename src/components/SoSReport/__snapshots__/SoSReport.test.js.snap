// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SoSReport renders entries when provided 1`] = `
Array [
  <h2
    className="ecl-u-type-heading-2"
  >
    Number of statements of support collected as reported by the organisers
  </h2>,
  <table
    className="eci-table ecl-u-type-paragraph"
  >
    <tbody>
      <tr>
        <td>
          Bulgaria
        </td>
        <td>
          12598
        </td>
      </tr>
      <tr>
        <td>
          Denmark
        </td>
        <td>
          4610
        </td>
      </tr>
    </tbody>
  </table>,
  <p
    className="ecl-u-type-paragraph-s"
  >
    *
     
    Certificates received after the time of submission. Figures not included in the total number of signatories.
  </p>,
]
`;

exports[`SoSReport renders signatures information when greater than 1 by totalSignatures 1`] = `
Array [
  <h2
    className="ecl-u-type-heading-2"
  >
    Number of statements of support collected as reported by the organisers
  </h2>,
  <table
    className="eci-table ecl-u-type-paragraph"
  >
    <tbody>
      <tr
        className="ecl-u-type-bold"
      >
        <td>
          Total number of signatories
        </td>
        <td>
          2
        </td>
      </tr>
    </tbody>
  </table>,
  <p
    className="ecl-u-type-paragraph-s"
  >
    *
     
    Certificates received after the time of submission. Figures not included in the total number of signatories.
  </p>,
]
`;
